lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -44,14 +44,40 @@ defmodule Mix.Tasks.Compile.App do
       The application environment is one of the most common ways
       to configure applications.
 
-  Let's see an example `application/0` function:
+    * `:start_phases` - specifies a list of phases and their arguments
+      to be called after the application is started and before any
+      included applications are started.
+
+    * `:included_applications` - specifies a list of applications
+      that will be included in the application. A process in an
+      included application considers itself belonging to the
+      primary application.
```

Comment:
We should be explicit that the application is responsible for starting the application's supervision tree inside its own, and that an application can only be included in one application on a node.